### ISSUE TITLE
Insufficient Permission for Persistent Volume Provisioning Example

### ISSUE THREAD

**Issue by jason-riddle** [kubernetes/kubernetes#23517]
I followed the instructions from the [persistent-volume-provisioning](https://github.com/kubernetes/kubernetes/blob/release-1.2/examples/experimental/persistent-volume-provisioning/README.md) example. When I run create, and then describe, I see the following.

```bash
# kubectl describe pv
Name:		pv-gce-18xko
Labels:		<none>
Status:		Failed
Claim:		default/claim-1
Reclaim Policy:	Delete
Access Modes:	RWO
Capacity:	1Gi
Message:	googleapi: Error 403: Insufficient Permission, insufficientPermissions
Source:
    Type:	GCEPersistentDisk (a Persistent Disk resource in Google Compute Engine)
    PDName:	placeholder-for-provisioning
    FSType:	ext4
    Partition:	0
    ReadOnly:	false
```

I'm using a GKE cluster which is running 1.2.0. Am I supposed to do anything else or is this not working as intended?

**Comment by dchen1107** [comment]
cc/ @roberthbailey

**Comment by roberthbailey** [comment]
/cc @kubernetes/goog-gke 

@saad-ali -- When creating a PV claim, which controller is responsible for talking to the GCE API (and where does it run)?

**Comment by saad-ali** [comment]
> @saad-ali -- When creating a PV claim, which controller is responsible for talking to the GCE API (and where does it run)?

Their are three controllers currently responsible for creating, binding, and deleting PV/PVC/volumes: provisioner, binder, and recycler. All are found [here](https://github.com/kubernetes/kubernetes/tree/master/pkg/controller/persistentvolume). All of them talk to the API server.

CC @kubernetes/sig-storage

**Comment by bprashanth** [comment]
>Their are three controllers currently responsible for creating, binding, and deleting PV/PVC/volumes: provisioner, binder, and recycler. All are found here. All of them talk to the API server.
CC @kubernetes/sig-storage

So the controllers run on the master, but I think the kubelet actually needs scope to use the gce_pd volume plugin (@saad-ali correct me if this is not the case)

**Comment by saad-ali** [comment]
Yes, the controllers run on master. So the master needs scope that will enable it to [create](https://cloud.google.com/compute/docs/reference/latest/disks/insert) and (in the future) [attach](https://cloud.google.com/compute/docs/reference/latest/instances/attachDisk) GCE PDs. Namely, `https://www.googleapis.com/auth/compute` and `https://www.googleapis.com/auth/cloud-platform`. The kubelet currently handles volume attachment, so nodes currently need the same scopes to attach GCE PDs to node.

**Comment by bprashanth** [comment]
Master scope is probably not the problem here. @jason-riddle can you check the scope on your nodes? You can ssh into a node and try `gcloud compute disks list` to make sure you have compute read, or run `gcloud container clusters describe foo` to see the oauthScopes.

**Comment by jason-riddle** [comment]
So I ran both commands (with sensitive info removed).

```bash
> ssh node-01
...
$ gcloud compute disks list
NAME                                      ZONE       SIZE_GB TYPE        STATUS
gke-kubernetes-cluster-XXXXXXX-node-c5kb us-east1-b 100     pd-standard READY
gke-kubernetes-cluster-XXXXXXX-node-l66n us-east1-b 100     pd-standard READY
gke-kubernetes-cluster-XXXXXXX-node-m3d9 us-east1-b 100     pd-standard READY
```

```bash
$ gcloud container clusters describe kubernetes-cluster
clusterIpv4Cidr: 10.0.0.0/14
...
currentMasterVersion: 1.2.0
currentNodeCount: 3
currentNodeVersion: 1.2.0
...
nodeConfig:
  ...
  oauthScopes:
  - https://www.googleapis.com/auth/compute
  - https://www.googleapis.com/auth/devstorage.read_only
  - https://www.googleapis.com/auth/logging.write
  - https://www.googleapis.com/auth/monitoring
nodeIpv4CidrSize: 24
...
status: RUNNING
zone: us-east1-b
```

**Comment by saad-ali** [comment]
@jason-riddle This looks like a GKE issue. We're looking into it.

**Comment by pwittrock** [comment]
I am guessing this is the issue.

<img width="305" alt="screen shot 2016-04-07 at 9 07 59 am" src="https://cloud.githubusercontent.com/assets/7882754/14358076/5459a352-fca0-11e5-96a1-612fa6d17be1.png">

**Comment by jason-riddle** [comment]
@pwittrock This matches the permissions I have with the following exception

```
Cloud Monitoring API     Full
```

**Comment by pwittrock** [comment]
That was from my GKE cluster, which also fails.  I believe the problem is the "Storage Read Only" permission.  I think there are 2 work arounds.  #1 is to manually provision the gce persistent disks as is shown in the [wordpress example](https://github.com/kubernetes/kubernetes/tree/master/examples/mysql-wordpress-pd).  I am currently working on changing the instance group template to give the nodes full storage permissions.  This is a much more invasive change as it requires restarting the instances.

**Comment by pwittrock** [comment]
Updating the templates didn't fix the issue.  @saad-ali do you have any updates?

**Comment by roberthbailey** [comment]
This is fixed by https://github.com/kubernetes/kubernetes/pull/23769 which has been cherry picked into the release branch and will be available once we cut the 1.2.2 release.

**Comment by pwittrock** [comment]
Great.  Thanks @roberthbailey

**Comment by saad-ali** [comment]
Closing. This has been fixed with Kubernetes v1.2.2 which should be available on GKE soon.
